storage/schema.org.md

```diff
@@ -0,0 +1,230 @@
+# REST API using Schema.org
+
+Request the root:
+
+```
+GET /files HTTP/2
+accept: application/ld+json
+```
+
+```
+HTTP/2 200
+content-type: application/ld+json
+
+{
+    "@context": "http://schema.org",
+    "@type": "WebAPI",
+    "potentialAction": [
+        {
+            "@type": "CreateAction",
+            "target": {
+                "@type": "EntryPoint",
+                "httpMethod": "POST",
+                "urlTemplate": "/files/actions",
+                "encodingType": "multipart/related; type=application/ld+json",
+                "contentType": "application/ld+json"
+            },
+            "agent-input": "required"
+            "result": {
+                "@type": "MediaObject",
+                "name-input": "required",
+                "isPartOf-input": "required",
+            }
+        }
+    ]
+}
+```
+
+Not sure of the best way of combining data and a file; other option is multi-step (ie create the action, then attach the file).
+
+---
+
+Perform a [`CreateAction`](https://schema.org/CreateAction):
+
+```
+POST /files/actions HTTP/2
+accept: application/ld+json
+content-type: multipart/related; boundary=foo_bar_baz; type=application/ld+json
+
+--foo_bar_baz
+content-type: application/ld+json
```

Comment:
so the `Content-Type` will never get more specific than this because the information is in the body?

Reply:
Could be possible to use a [`profile` parameter](https://www.w3.org/TR/json-ld11/#iana-considerations) to provide more information about what's inside, but never done it myself. (@stephenwf mentioned this the other week.)

Reply:
Yeh although its generally used only as part of the content negotiation, so an unsupported profile passed to the API provider will throw. Useful if you have versioned profiles (like IIIF) and want to opt-in to support future or past versions